sdk/python/lib/pulumi/dynamic/dynamic.py

```diff
@@ -15,7 +15,7 @@
 import asyncio
 import base64
 import pickle
-from typing import Any, Optional, List, TYPE_CHECKING
+from typing import Any, Optional, List, TYPE_CHECKING, no_type_check, cast
```

Comment:
when would ew use no_type_check (versus just casting to the type we expect).

Reply:
Mypy has additional rules for what it considers safe/unsafe behavior, outside of just type checking. I use the `@no_type_check` decorator here to disable type checking on the whole method: https://github.com/pulumi/pulumi/pull/3758/files/9008ac25d834e04a643a1d96b45d6122183af1be#diff-47c43a5ad2821b986b45591caf8593b8R205-R206

One of the lines there does something method assignment, which mypy doesn't allow. We already have a `pylint: disable` comment on that line, so we can't inline the ignore. Mypy only supports line-level, function-level, and file-level ignores. No support for doing code blocks.